src/lib/isTime.js

```diff
@@ -0,0 +1,23 @@
+import merge from './util/merge';
+
+const default_time_options = {
+  hourFormat: 'hour24',
```

Comment:
Hi, personally I don't find the names of these options correct. Why not `24hour` and `12hour`?

Reply:
Yeah, it's weird! Your sugestion is more readable...

Reply:
Indeed I remember why I choose theses names, acording to docs of [toLocaleTimeString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleTimeString) function there's a parameter called `hour12`, so to be like this pattern I picked hour12 and hour24.

Reply:
Yes in that case it makes more sense because the name of an object property starting with a number can only be used with quotes.
```js
const obj = {
  '123foo': true,
  foo123: false,
}
```
What we may do to reflect `toLocaleTimeString` behaviour is to use a boolean `hour12` option that can be set to false if we want `hour24` format:
```js
const options = {
  hour12: true // defaults to 12
}
```

Reply:
Make sense, I'll work on this.

Reply:
In my view, despite verbosity the current format avoid cognitive effort, personaly I have problems with a bunch of boolean parameters. They explicity say what means, like below: 

`
const options = {
   hourFormat: hour24,
   mode: withSeconds 
}
`

In `withSeconds` parameter makes more sense to be a boolean...

Reply:
Let's see @profnandaa and @tux-tn opinion on this.

Reply:
Sorry for the very long delay (An irony in a PR about time validation 😅 )
I suggest keeping the same options logic used in other validators:
```suggestion
const default_time_options = {
  format: 24, // Can be one of  24 or 12
  allow_seconds: true
};
```

The idea is to keep things simple and self-explanatory, what do you think guys?

Reply:
I still prefer a boolean flag for hours 🤔

Reply:
I don't know, both ideas are personal opinions. Let's see @profnandaa feedback :man_shrugging: 
What do you think about `allow_seconds` ? (We are already using snake case in most of the options)

Reply:
It is the best choice compared to `mode: 'withSeconds'` IMO